ci-operator/step-registry/smongiar/run-tests/smongiar-run-tests-commands.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+function create_dc_and_xpaas_pod()
+{
+  oc create -f - <<EOF
+kind: DeploymentConfig
+apiVersion: apps.openshift.io/v1
+metadata:
+  name: xpaas-qe
+  labels:
+    application: xpaas-qe
+spec:
+  strategy:
+    type: Recreate
+  triggers:
+    - type: ImageChange
+      imageChangeParams:
+        automatic: true
+        containerNames:
+          - xpaas-qe
+        from:
+          kind: ImageStreamTag
+          name: xpaas-qe:latest
+    - type: ConfigChange
+  replicas: 1
+  selector:
+    deploymentConfig: xpaas-qe
+  template:
+    metadata:
+      name: xpaas-qe
+      labels:
+        deploymentConfig: xpaas-qe
+        application: xpaas-qe
+    spec:
+      securityContext:
+        runAsUser: 0
+      terminationGracePeriodSeconds: 60
+      volumes:
+        - name: mvn-settings
+          configMap:
+            name: mvn-settings
+            items:
+            - key: settings.xml
+              path: custom.settings.xml
+        - name: test-properties
+          configMap:
+            name: test-properties
+            items:
+            - key: test.properties
+              path: test.properties
+        - name: xpaas-qe-volume
+          persistentVolumeClaim:
+            claimName: persistent-xpaas-qe
+      containers:
+        - name: xpaas-qe
+          ports:
+            - containerPort: 22
+              protocol: TCP
+            - containerPort: 80
+              protocol: TCP
+            - containerPort: 443
+              protocol: TCP
+          env:
+          - name: MVN_ARGS
+            value: -Dlpinterop
+          - name: MVN_SETTINGS_PATH
+            value: /tmp/custom.settings.xml
+          - name: MVN_PROFILES
+            value: test-fuse,openshift4-current
+          - name: TEST_EXPR
+            value: SpringBootCamelXaFabricTest
+          - name: OPENSHIFT_NAMESPACE
+            value: smongiar
+          - name: NAMESPACE
+            value: smongiar
+          - name: NAMESPACE_PREFIX
+            value: fuse
+          - name: SUREFIRE_REPORTS_FOLDER
+            value: /deployments/xpaas-qe/test-fuse/target/surefire-reports
+          volumeMounts:
+            - name: mvn-settings
+              mountPath: /tmp/
+            - name: test-properties
+              mountPath: /mnt/
+            - name: xpaas-qe-volume
+              mountPath: /tmp/xtf-oc-cache/
+              subPath: xtf-oc-cache
+            - name: xpaas-qe-volume
+              mountPath: /tmp/surefire-reports/
+              subPath: surefire-reports
+            - name: xpaas-qe-volume
+              mountPath: /tmp/log/
+              subPath: log
+            - name: xpaas-qe-volume
+              mountPath: /deployments/.m2/
+              subPath: maven-repo
+EOF
+}
+
+echo "Running the tests after deployment-config and xpaas-qe pod creation"
+create_dc_and_xpaas_pod
```

Comment:
@smongiar we also need to artifact the xmls and logs generated. Please add the `cp` command to copy the xmls and other logs after running tests to `${ARTIFACT_DIR}/`.

Reply:
@smongiar yes you can get access to test cluster by adding the wait ref like [this](https://github.com/openshift/release/pull/36871/commits/94b87061d6bc1e7854048ddedf1ad9076ebe2bb1), and the cluster will only be up for the `SLEEP_DURATION` period. You can add this after test execution step.
And when the pod will be running for this ref you can access the cluster by following [these](https://cspi-qe.gitbook.io/ocp-ci-onboarding/onboarding/onboarding_guide/scenario_development_guide#accessing-the-test-cluster) steps.